lib/tools.js

```diff
@@ -1847,14 +1847,65 @@ function encrypt(key, value) {
  * @param {string} value - value to decrypt
  * @returns {string}
  */
-function decrypt(key, value) {
+function decryptLegacy(key, value) {
     let result = '';
     for(let i = 0; i < value.length; i++) {
         result += String.fromCharCode(key[i % key.length].charCodeAt(0) ^ value.charCodeAt(i));
     }
     return result;
 }
 
+/**
+ * encrypts a value by a given key via AES-256-CBC
+ *
+ * @param {string} key - Secret key
+ * @param {string} value - value to decrypt
+ * @returns {string}
+ */
+function encrypt(key, value) {
+    if (key.length < 32) {
+        // key is too short for AES-256-CBC - fallback to old encryption
+        return encryptLegacy(key, value);
+    }
+
+    crypto = crypto || require('crypto');
+
+    key = key.substr(0, 32);
```

Comment:
means that code that the key needs to be exact 32 chars ... or only first 32 chars are used?

Reply:
We only use first 32 chars of the secret, because secret is too long.

Reply:
How long is the secret and which characters does it contain? 32 ascii chars seems a bit weak for a 256-bit key.

Reply:
You are right. I will look into it later.

Reply:
secrets are generated with 192bits in https://github.com/ioBroker/ioBroker.js-controller/blob/master/lib/setup/setupSetup.js#L190

Thus, I changed encryption to aes 192 cbc so we can use full secret. Changing the secret will most likely be breaking and 192 bit should be fine.